araproc/analysis/daq_quality_cut.py

```diff
@@ -0,0 +1,253 @@
+    channel_mask : array-like
+        The channel mask for the event.
+
+    Returns
+    -------
+    bool
+        True if there are DAQ structure errors, False otherwise.
+    """
+
+    num_chs = constants.rf_chans_per_dda   # Number of channels
+    num_blocks_per_dda = constants.num_blocks_per_dda
+    bi_ch_mask = 1 << np.arange(num_chs, dtype=int)  # Bit mask for channels
+    dda_ch = np.arange(num_ddas, dtype=int)  # DDA channels
+    dda_idx = (channel_mask & 0x300) >> 8  # DDA index from channel mask
+    max_blk_diff = -(num_blocks_per_dda - 1)  # Maximum block difference (wrap-around at 512)
+
+    daq_st_err = np.full(5, 0, dtype=int)  # Error array for this single event
+
+    # Block index for this event
+    blk_idx_evt = np.asarray(irs_block_number, dtype=int)
+
+    # Check for block length errors (remainder when divided by number of DDAs)
+    daq_st_err[0] = len(blk_idx_evt) % num_ddas
+    if daq_st_err[0] != 0:
+        return True
+
+    # Reshape block indices into groups of DDAs for comparison
+    blk_idx_reshape = np.reshape(blk_idx_evt, (-1, num_ddas))
+    daq_st_err[1] = int(np.any(blk_idx_reshape != blk_idx_reshape[:, 0][:, np.newaxis]))
+
+    for dda in range(num_ddas):
+        blk_idx_dda = blk_idx_reshape[:, dda]
+        first_block_idx = blk_idx_dda[0]
+        last_block_idx = blk_idx_dda[-1]
+        block_diff = len(blk_idx_dda) - 1
+
+        # Check block continuity, accounting for wrap-around at 512
+        if first_block_idx + block_diff != last_block_idx:
+            if num_blocks_per_dda - first_block_idx + last_block_idx != block_diff:
+                daq_st_err[2] += 1
+
+        # Check block index increments
+        blk_diff = np.diff(blk_idx_dda).astype(int)
+        incre_flag = np.any(np.logical_and(blk_diff != 1, blk_diff != max_blk_diff))
+        if incre_flag:
+            daq_st_err[2] += 1
+
+    # Check DDA index consistency
+    dda_idx_evt = np.asarray(dda_idx, dtype=int)
+    dda_idx_reshape = np.reshape(dda_idx_evt, (-1, num_ddas))
+    daq_st_err[3] = int(np.any(dda_idx_reshape != dda_ch[np.newaxis, :]))
+
+    # Check channel mask consistency
+    ch_mask_evt = np.asarray(channel_mask, dtype=int)
+    ch_mask_reshape = np.repeat(ch_mask_evt[:, np.newaxis], num_chs, axis=1)
+    ch_mask_bit = ch_mask_reshape & bi_ch_mask[np.newaxis, :]
+    daq_st_err[4] = int(np.any(ch_mask_bit != bi_ch_mask[np.newaxis, :]))
+
+    # Return True if there are any DAQ structure errors
+    return np.sum(daq_st_err) > 0
+
+
+def get_read_win_limit(st, run):
+
+    """
+    Detect readout window errors for a single event based on event data.
+
+    Parameters
+    ----------
+    st : int
+        Station id number.
+    run : int
+        Run Number
+    Returns
+    -------
+    rf_readout_limit: int
+        RF trig readout window limit.
+    soft_readout_limit: int
+        Soft trig readout window limit.
+    """
+
+    file = pkg_resources.open_text(config_files,"analysis_configs.yaml")
```

Comment:
This is awesome! Can you update the yaml files with some placeholder values for folks to update?

Reply:
Haha, the irony is I thought I pushed the yaml file as well.

Reply:
haha no worries! let's just make sure to add it before merging :)